libs/langgraph/langgraph/pregel/__init__.py

```diff
@@ -789,19 +982,44 @@ def _defaults(
         stream_mode = stream_mode if stream_mode is not None else self.stream_mode
         if not isinstance(stream_mode, list):
             stream_mode = [stream_mode]
+
+        if config and config.get("configurable", {}).get(CONFIG_KEY_CHECKPOINTER):
+            parent_checkpointer = config["configurable"][CONFIG_KEY_CHECKPOINTER]
+        else:
+            parent_checkpointer = None
+
         if config and config.get("configurable", {}).get(CONFIG_KEY_READ) is not None:
             # if being called as a node in another graph, always use values mode
             stream_mode = ["values"]
+
+            if (
+                (interrupt_before or interrupt_after)
+                and parent_checkpointer is not None
+                and self.checkpointer is None
+            ):
+                raise ValueError(
```

Comment:
ah, so you're saying we should always inherit it automatically if we use interrupts and otherwise just add it via `INHERIT_CHECKPOINTER` as an opt-in?